wellknownimports/wellknownimports.go

```diff
@@ -18,14 +18,58 @@ package wellknownimports
 
 import (
 	"embed"
+	"fmt"
 	"io"
+	"io/fs"
+	"path/filepath"
 
 	"github.com/bufbuild/protocompile"
 )
 
 //go:embed google/protobuf/*.proto google/protobuf/*/*.proto
 var files embed.FS
 
+var filesByName = make(map[string]string)
+
+func init() {
+	// We initialize a map instead of going through the embed.FS for two reasons:
+	//
+	// 1. ReadFile performs an unnecessary copy of the file's text.
+	// 2. It requires some non-trivial work to identify when the user passes a
+	//    directory such as "google/" or "google/protobuf/".
+	err := fs.WalkDir(files, ".", func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if !d.IsDir() {
+			bytes, err := files.ReadFile(path)
+			if err != nil {
+				return err
+			}
+
+			path = filepath.Clean(path)
+			if filepath.Separator != '/' {
```

Comment:
Is this actually an issue on windows? Seems like this can be simplified to https://github.com/golang/go/issues/68909#issuecomment-2295808381